src/main/java/uk/gov/hmcts/dts/fact/util/AddressType.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.dts.fact.util;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@Getter
+public enum AddressType {
+    VISIT_US("Visit us", true),
+    WRITE_TO_US("Write to us", false),
+    VISIT_OR_CONTACT_US("Visit or contact us", true);
+
+    private static final Map<String, AddressType> LOOKUP = new ConcurrentHashMap<>();
+
+    private final String name;
+    private final boolean isCourtAddress;
+
+    static {
+        Arrays.stream(values())
+            .forEach(t -> LOOKUP.put(t.name.toLowerCase(Locale.getDefault()), t));
+    }
+
+    public static AddressType findByName(final String name) {
+        final String key = name.toLowerCase(Locale.getDefault());
+        if (!LOOKUP.containsKey(key)) {
+            throw new IllegalArgumentException("Unknown address type: " + name);
```

Comment:
Minor again, can use string.format rather than + if you want...I don't really mind though up 2 u

Reply:
https://redfin.engineering/java-string-concatenation-which-way-is-best-8f590a7d22a8 - it says concatenation is better :) I normally avoid using string.format if we don't need to use format specifier. I think in this case, a string concatenation is sufficient.